addon/Feeds/TopSitesFeed.js

```diff
@@ -23,28 +26,27 @@ module.exports = class TopSitesFeed extends Feed {
       // Get metadata from PreviewProvider
       links = yield this.options.getCachedMetadata(links, "TOP_FRECENT_SITES_RESPONSE");
 
+      this.missingData = false;
+
       // Get screenshots if the favicons are too small
       if (experiments.screenshots) {
-        try {
-          links = yield getScreenshots(links, site => {
-            if (site.favicon_height >= 96 && site.favicon_width >= 96 && isRootDomain(site.url)) {
-              // If we are at the "root domain path" and the icon is big enough,
-              // we don't show a screenshot.
-              return false;
-            }
-            return true;
-          });
-          links = links.map(link => {
-            if (link.screenshot) {
+        for (const link of links) {
```

Comment:
Sorry for the random drop-in:

This is actually totally valid syntax for `for...of`, since `const link` is scoped to the inner block of the loop and not the outer block.

However, because of a bug in Firefox (https://bugzilla.mozilla.org/show_bug.cgi?id=1101653)  this throws a syntax error. It's fixed in 51+, but anything lower than that still hast the bug...

... so, maybe you might what to change it to `let` anyway. 🤓